sys/aarch64/gpio.c

```diff
@@ -0,0 +1,94 @@
+#include <sys/rman.h>
+#include <sys/bus.h>
+#include <aarch64/bcm2835reg.h>
+#include <aarch64/bcm2835_gpioreg.h>
+
+/* The function select registers are used to define the operation of the
+ * general-purpose I/O pins. */
+#define GPFSEL(x) ((x) * sizeof(uint32_t))
+
+#define GPFSEL_PINS_PER_REGISTER 10
+#define GPFSEL_BITS_PER_PIN 3
+
+/* The output set registers are used to set a GPIO pin. */
+#define GPSET(x) (0x1C + (x) * sizeof(uint32_t))
+
+/* The output clear registers are used to clear a GPIO pin. */
+#define GPCLR(x) (0x28 + (x) * sizeof(uint32_t))
+
+/* The pin level registers return the actual value of the pin. */
+#define GPLEV(x) (0x34 + (x) * sizeof(uint32_t))
+
+/* The event detect status registers are used to record level and edge events
+ * on the GPIO pins. */
+#define GPEDS(x) (0x40 + (x) * sizeof(uint32_t))
+
+/* The rising edge detect enable registers define the pins for which a rising
+ * edge transition sets a bit in the event detect status registers GPEDS. */
+#define GPREN(x) (0x4C + (x) * sizeof(uint32_t))
+
+/* The falling edge detect enable registers define the pins for which a
+ * falling edge transition sets a bit in the event detect status registers
+ * GPEDS. */
+#define GPFEN(x) (0x58 + (x) * sizeof(uint32_t))
+
+/* The high level detect enable registers define the pins for which a high
+ * level sets a bit in the event detect status register GPEDS. */
+#define GPHEN(x) (0x64 + (x) * sizeof(uint32_t))
+
+/* The low level detect enable registers define the pins for which a low level
+ * sets a bit in the event detect status register GPEDS. */
+#define GPLEN(x) (0x70 + (x) * sizeof(uint32_t))
+
+/* The asynchronous rising edge detect enable registers define the pins for
+ * which a asynchronous rising edge transition sets a bit in the event detect
+ * status registers GPEDS. */
+#define GPAREN(x) (0x7C + (x) * sizeof(uint32_t))
+
+/* The asynchronous falling edge detect enable registers define the pins for
+ * which a asynchronous falling edge transition sets a bit in the event detect
+ * status registers GPEDS. */
+#define GPAFEN(x) (0x88),
```

Comment:
This one is different that the others. Is that ok?

Reply:
Fixed.